src/coordinator_service.rs

```
@@ -795,6 +799,43 @@ pub struct CoordinatorServer {
     addr: SocketAddr,
     coordinator: Arc<Coordinator>,
     shared_state: Arc<state::App>,
+    config: Arc<ServerConfig>,
+    server_handle: axum_server::Handle,
+}
+
+async fn metrics_handler(
+    State(app): State<Arc<state::App>>,
+) -> Result<axum::response::Response<axum::body::Body>, IndexifyAPIError> {
+    let metric_families = app.metrics.registry.gather();
+    let mut buffer = vec![];
+    let encoder = prometheus::TextEncoder::new();
+    encoder.encode(&metric_families, &mut buffer).unwrap();
+
+    Ok(axum::response::Response::new(axum::body::Body::from(
+        buffer,
+    )))
+}
+
+fn start_server(app: &CoordinatorServer) -> JoinHandle<Result<()>> {
+    let server = axum::Router::new()
+        .route("/metrics", get(metrics_handler))
+        .with_state(app.shared_state.clone());
+    let addr: SocketAddr = format!(
+        "{}:{}",
+        app.config.listen_if, app.config.coordinator_http_port
+    )
+    .parse()
+    .unwrap();
```

Unwrap is probably not going to work here. Can we return an error and make the startup fail completely. Please test by specifying something wrong.

---

added check